File: Projeto_6/projeto 6 celso/src/usuario.py
# src/usuario.py
from .tipo_usuario import TipoUsuario

class Usuario:
    def __init__(self, nome: str, id_usuario: int, tipo: TipoUsuario):
        self._nome = nome
        self._id_usuario = id_usuario
        self._tipo = tipo

    @property
    def nome(self) -> str:
        return self._nome

    @property
    def id_usuario(self) -> int:
        return self._id_usuario

    @property
    def tipo(self) -> TipoUsuario:
        return self._tipo

File: projeto-8/Usuario.cpp
#include "Usuario.h"

Usuario::Usuario(std::string nome, std::string email, TipoUsuario tipo)
    : nome(nome), email(email), tipo(tipo) {
}

const std::string& Usuario::getNome() const {
    return this->nome;
}

const std::string& Usuario::getEmail() const {
    return this->email;
}
